EKF/control.cpp

```diff
@@ -1079,6 +1079,11 @@ void Ekf::controlHeightFusion()
 			}
 		}
 
+		if (_control_status.flags.rng_hgt && _range_sensor.isDataHealthy()) {
```

Comment:
Can you replace 1082 - 1095 with this block?
```
               if (_control_status.flags.ev_hgt && _ev_data_ready) {
                       fuse_height = true;
               } else if (_control_status.flags.rng_hgt && _range_sensor.isDataHealthy()) {
                       fuse_height = true;
               } else if (_control_status.flags.baro_hgt && _baro_data_ready && !_baro_hgt_faulty) {
                        fuse_height = true;
               }

```

This groups the statements together like for GPS (few lines above), and adds a check for `_ev_data_ready` which fixes the issue you saw here https://github.com/PX4/PX4-ECL/pull/994#discussion_r621816867.

After that you can also revert the change you made on line 792 in #994, and timeouts will work as designed.